### DISCUSSION TITLE
How can I make dagster fast for small data?

### DISCUSSION THREAD

**Discussion by brentshulman-silkline** [dagster-io/dagster#20672] [Q&A] [answered]
Hello all! My team is working on an integration pipeline in Dagster that we would like to run fast - ideally, under 1 min fast. Right now, the bottleneck is a couple things:

1. IO
2. Process Startup

With IO, we think that we can get what we need using the `mem_io_manager`

With the process startup, the `multiprocess_executor` gives us parallelism, but the overhead of each process is very slow. It increases our overall pipeline time by 100% per node. The `in_process` executor solves that problem, but we lose the parallelism. 

Any ideas on an approach to solve this?

**Comment by sryza** [comment, marked as answer]
Hi @brentshulman-silkline  - this issue tracks adding an in-process multi-threaded executor, but we don't yet have this capability: https://github.com/dagster-io/dagster/issues/4041. This would mainly help with IO parallelism, because the Python GIL limits CPU parallelism.
